crates/runtimes/src/outputs.rs

```diff
@@ -0,0 +1,463 @@
+        let row_heights = (self.num_rows() as f32 * 1.2) + 1.0;
+
+        (row_heights as u8).saturating_add(2)
+    }
+}
+
+impl OutputType {
+    fn render(&self, cx: &ViewContext<ExecutionView>) -> Option<AnyElement> {
+        let el = match self {
+            // Note: in typical frontends we would show the execute_result.execution_count
+            // Here we can just handle either
+            Self::Plain(stdio) => Some(stdio.render(cx)),
+            // Self::Markdown(markdown) => Some(markdown.render(theme)),
+            Self::Stream(stdio) => Some(stdio.render(cx)),
+            Self::Image(image) => Some(image.render(cx)),
+            Self::Message(message) => Some(div().child(message.clone()).into_any_element()),
+            Self::Table(table) => Some(table.render(cx)),
+            Self::ErrorOutput {
+                ename,
+                evalue,
+                traceback,
+            } => render_error_output(ename, evalue, traceback, cx),
+        };
+
+        el
+    }
+}
+
+impl LineHeight for OutputType {
+    /// Calculates the expected number of lines
+    fn num_lines(&self, cx: &mut WindowContext) -> u8 {
+        match self {
+            Self::Plain(stdio) => stdio.num_lines(cx),
+            Self::Stream(stdio) => stdio.num_lines(cx),
+            Self::Image(image) => image.num_lines(cx),
+            Self::Message(message) => message.lines().count() as u8,
+            Self::Table(table) => table.num_lines(cx),
+            Self::ErrorOutput {
+                ename,
+                evalue,
+                traceback,
+            } => {
+                let mut height: u8 = 0;
+                height = height.saturating_add(ename.lines().count() as u8);
+                height = height.saturating_add(evalue.lines().count() as u8);
+                height = height.saturating_add(traceback.num_lines(cx));
+                height
+            }
+        }
+    }
+}
+
+fn render_error_output(
+    ename: &String,
+    evalue: &String,
+    traceback: &TerminalOutput,
+    cx: &ViewContext<ExecutionView>,
+) -> Option<AnyElement> {
+    let theme = cx.theme();
+
+    let colors = cx.theme().colors();
+
+    Some(
+        v_flex()
+            .w_full()
+            .bg(colors.background)
+            .p_4()
+            .border_l_1()
+            .border_color(theme.status().error_border)
+            .child(
+                h_flex()
+                    .font_weight(FontWeight::BOLD)
+                    .child(format!("{}: {}", ename, evalue)),
+            )
+            .child(traceback.render(cx))
+            .into_any_element(),
+    )
+}
+
+#[derive(Default)]
+pub enum ExecutionStatus {
+    #[default]
+    Unknown,
+    #[allow(unused)]
+    ConnectingToKernel,
+    Executing,
+    Finished,
+}
+
+pub struct ExecutionView {
+    pub execution_id: ExecutionId,
+    pub outputs: Vec<OutputType>,
+    pub status: ExecutionStatus,
+}
+
+pub fn svg_to_vec(text: &str, scale: f32) -> Result<OutputType> {
```

Comment:
I had the impression that the GPUI image view can already render SVGs and SVG files are just forced to use the text editor instead of the image preview because people want to edit them. Could that functionality be reused here instead of reimplementing it?

Reply:
The SVG renderer is based on reading a path. I should probably drop SVG support entirely from this PR because I've disabled it above due to not having full SVG features enabled, like text:

![image](https://github.com/zed-industries/zed/assets/836375/2208ae00-6d58-49f6-917c-7961877ed7e6)

Reply:
Yeah, shipping the initial version without SVG and later improving the already existing image view to handle the needed SVG features sounds like a preferrable approach to me.